### PR TITLE
Respect repo default branch

### PR DESCRIPTION
**Pull request by sgugger**
We hardcode `"master"` for the version in development but we should use the repo default branch (for instance `"main"`). This PR addresses that.

### CODE REVIEW

File: src/doc_builder/commands/build.py [hunk 1]
```diff
@@ -66,6 +66,22 @@ def locate_kit_folder():
...
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            check=True,
+            encoding="utf-8",
+            cwd=repo_folder,
+        )
+        branches = p.stdout.strip()
+        return "master" if "master" in branches else "main"
+    except Exception:
+        return "main"
```
Comment (mishig25) [inline src/doc_builder/commands/build.py:82]: why is main returned on exception?

**Comment by mishig25** [comment]
Update here to support main as well:
from:
```
if version == "master": 
```
to:
```
if version in ["master", "main"]: 
```
https://github.com/huggingface/doc-builder/blob/2e0f6e78eaf6fe3d02259769f23169bdf370bd4d/src/doc_builder/utils.py#L24-L30

**Review by mishig25 — APPROVED** [review]
Lgtm!

File: src/doc_builder/commands/build.py (hunk 1 above)
Comment (LysandreJik) [reply src/doc_builder/commands/build.py:82]: If you want to do it perfectly you could parse it from the result of `git symbolic-ref refs/remotes/origin/HEAD`, but it's fine for me to take `main` as the default.

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Pull request merged**